static/js/apps/eval_embeddings/app.tsx

```diff
@@ -15,134 +15,139 @@
-      (embeddings) => {
-        setCustomDescription(embeddings);
-      }
-    );
+    fetchEmbeddings(
+      lines,
+      Object.values(props.indexes).map((x) => x.model)
+    ).then((embeddings) => {
+      setDescription(embeddings);
+    });
+  };
+
+  const processQueries = (queryInput: string) => {
+    setQueries(queryInput.split("\n"));
   };
 
-  const handleCustomQuery = (event: React.FormEvent<HTMLFormElement>) => {
+  useEffect(() => {
+    processDescription(DEFAULT_DESCRIPTION);
+    processQueries(DEFAULT_QUERIES);
+  }, []);
+
+  const handleDescription = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const form = event.currentTarget;
+    const overrideInput = form.override ? form.override.value : "";
+    processDescription(overrideInput);
+  };
+
+  const handleQuery = (event: React.FormEvent<HTMLFormElement>) => {
     event.preventDefault();
     const form = event.currentTarget;
     const queryInput = form.query.value;
-    const svInput = form.statVars ? form.statVar.value : "";
-    setCustomQuery(queryInput);
-    setCustomGolden(svInput.split(",").map((s) => s.trim()));
+    processQueries(queryInput);
   };
 
-  const handleUpdateOverallScore = (
-    model: string,
-    sentence: string,
-    score: number
+  const handleFileUpload = (
+    event: React.ChangeEvent<HTMLInputElement>,
+    type: "description" | "query",
+    textArea: HTMLTextAreaElement
   ) => {
-    if (overallScore.current[model] === undefined) {
-      overallScore.current[model] = {};
+    const file = event.target.files?.[0];
+    if (file) {
+      const reader = new FileReader();
+      reader.onload = (e) => {
+        const text = e.target?.result as string;
+        if (type === "description") {
+          processDescription(text);
+        } else if (type === "query") {
+          processQueries(text);
+        }
+        textArea.value = text;
+      };
+      reader.readAsText(file);
     }
-    overallScore.current[model][sentence] = score;
-    for (const modelName in overallScore.current) {
-      if (
-        Object.values(overallScore.current[modelName]).length !==
-        Object.keys(props.evalGolden).length
-      ) {
-        return;
-      }
-    }
-    // Only update when all models have the scores for all the eval queries.
-    setCompletedOverallScore(overallScore.current);
   };
 
   return (
-    <div>
+    <>
       <div className="app-section">
-        <h3>Try Your Own Stat Var Descriptions</h3>
-        <p>
-          These descriptions override existing descriptions and takes effect in
-          all the results below
-        </p>
-        <p>
-          Each row is in the form of
-          &quot;stat_var_dcid,description1;description2;...&quot;
-        </p>
-        <form onSubmit={handleCustomDescription}>
-          <textarea name="override" />
+        <h3>
+          Enter or upload stat var descriptions (each row like
+          &quot;dcid,desc1;desc2&quot;)
```

Comment:
To be safe against `,` within desc1 and desc2, maybe do:

`dcid,"desc1;desc2"`

?